BAZProjectC1/Components/CarosuelMenu/CarosuelMenu.swift

```diff
@@ -0,0 +1,115 @@
+//
+//  CarosuelMenu.swift
+//  BAZProjectC1
+//
+//  Created by 1044336 on 23/09/22.
+//
+
+import UIKit
+
+struct CarosuelMenuConfiguration {
+    let frame: CGRect
+    let optionsTitles: [String]
+    let itemBackgroundColor: UIColor
+    let itemBorderBackgroundColor: UIColor
+    let itemSelectedBackgroundColor: UIColor
+    let itemSelectedBorderBackgroundColor: UIColor
+    let itemBorder: Double
+    
+    init(frame:CGRect,
+         optionsTitles:[String] = [],
+         itemBackgroundColor: UIColor = .clear,
+         itemBorderBackgroundColor: UIColor = .clear,
+         itemSelectedBackgroundColor: UIColor = .clear,
+         itemSelectedBorderBackgroundColor: UIColor = .clear,
+         itemBorder: Double = 10.0) {
+        
+        self.frame = frame
+        self.optionsTitles = optionsTitles
+        self.itemBackgroundColor = itemBackgroundColor
+        self.itemBorderBackgroundColor = itemBorderBackgroundColor
+        self.itemSelectedBackgroundColor = itemSelectedBackgroundColor
+        self.itemSelectedBorderBackgroundColor = itemSelectedBorderBackgroundColor
+        self.itemBorder = itemBorder
+    }
+}
+
+class CarosuelMenu: UIView {
+    public weak var delegate: CarosuelMenuDelegate?
+    private(set) var itemBackgroundColor: UIColor = .clear
+    private(set) var itemBorderBackgroundColor: UIColor = .clear
+    private(set) var itemSelectedBackgroundColor: UIColor = .clear
+    private(set) var itemSelectedBorderBackgroundColor: UIColor = .clear
+    private(set) var itemBorder: Double = 10.0
+    private(set) lazy var menuOptions: [String] = {
+        return []
+    }()
+    
+    public var optionSelected: Int = 0{
```

Comment:
¿Es necesario que esta propiedad sea publica? De ser así, podría tener un private(set)
```suggestion
    private(set) var optionSelected: Int = 0 {
```